node/config.go

```diff
@@ -394,7 +400,7 @@ func (c *Config) NodeKey() *ecdsa.PrivateKey {
 	if err != nil {
 		log.Crit(fmt.Sprintf("Failed to generate node key: %v", err))
 	}
-	instanceDir := filepath.Join(c.DataDir, c.name())
+	instanceDir := c.DataDir
```

Comment:
It looks like it used to be that `keyfile` for LoadECDSA and for `SaveECDSA` were different due to the inclusion of `c.name()` in the `instanceDir`, and now it can be the same. I don't know the history behind this or what the behavior should be.

Reply:
Actually, keyfile's directory and instanceDir were same before too. 
This is ensured by keyfile being assigned to`ResolvePath`'s return value i.e  `filepath.Join(c.instanceDir(), path)`